compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4276,6 +4278,30 @@ public void visit(BLangServiceConstructorExpr serviceConstructorExpr) {
         result = rewriteExpr(typeInit);
     }
 
+    @Override
+    public void visit(BLangStreamConstructorExpr streamConstructorExpr) {
+        BLangInvocation streamConstructInvocation = streamConstructInvocation(streamConstructorExpr);
+        result = rewriteExpr(streamConstructInvocation);
+    }
+
+    private BLangInvocation streamConstructInvocation(BLangStreamConstructorExpr streamConstructorExpr) {
+        BLangLambdaFunction lambdaFunction = streamConstructorExpr.lambdaFunction;
+        BInvokableSymbol symbol = (BInvokableSymbol) symTable.langStreamModuleSymbol.scope
+                .lookup(Names.CONSTRUCT_STREAM).symbol;
+
+        BType targetType = ((BStreamType) streamConstructorExpr.type).constraint;
+        BType typedescType = new BTypedescType(targetType, symTable.typeDesc.tsymbol);
+        BLangTypedescExpr typedescExpr = new BLangTypedescExpr();
+        typedescExpr.resolvedType = targetType;
+        typedescExpr.type = typedescType;
+
+        BLangInvocation streamConstructInvocation = ASTBuilderUtil.createInvocationExprForMethod(
+                streamConstructorExpr.pos, symbol, new ArrayList<>(Lists.of(typedescExpr, lambdaFunction)),
+                symResolver);
+        streamConstructInvocation.type = symTable.streamType;
```

Comment:
dont we need to set the constraint type of the stream type here?.. meaning that we have to create a new stream type, without using symTable.streamType

Reply:
At this point, the constraint would be a `pureType`.